packages/eslint-plugin/src/util/getOperatorPrecedence.ts

```diff
@@ -0,0 +1,347 @@
+import { SyntaxKind } from 'typescript';
+
+export enum OperatorPrecedence {
+  // Expression:
+  //     AssignmentExpression
```

Comment:
@bradzacher
Precedence now determines whether parentheses need to be added or not.

I copied the typescript code verbatim, is that ok?
https://github.com/microsoft/TypeScript/blob/af689cc5d527789cbc1d18a7b8aae6fa46051ea7/src/compiler/utilities.ts#L3446-L3783

Reply:
not a problem at all - we *have* to copy it because it's a `const` enum!